samples/csharp/ReferenceApp/ReferenceApp/ReferenceApp.cs

```diff
@@ -81,7 +81,53 @@ private static async void App(Options options)
             {
                 logger.LogInformation("using DynamoDB-based metastore...");
                 AWSConfigs.AWSRegion = "us-west-2";
-                metastore = DynamoDbMetastoreImpl.NewBuilder().Build();
+                DynamoDbMetastoreImpl.Builder builder = DynamoDbMetastoreImpl.NewBuilder();
+
+                if (!string.IsNullOrEmpty(options.DynamodbRegion))
+                {
+                    if (!string.IsNullOrEmpty(options.DynamodbEndpoint) ||
+                        !string.IsNullOrEmpty(options.DynamodbSigningRegion))
+                    {
+                        logger.LogError("Either provide a DynamoDb region OR Endpoint with signing region");
+                        Console.WriteLine(HelpText.AutoBuild(cmdOptions, null, null));
+                        return;
+                    }
+
+                    builder.WithRegion(options.DynamodbRegion);
+                }
+
+                if (!string.IsNullOrEmpty(options.DynamodbEndpoint) ||
+                    !string.IsNullOrEmpty(options.DynamodbSigningRegion))
+                {
+                    if (!string.IsNullOrEmpty(options.DynamodbRegion))
+                    {
+                        logger.LogError("Either provide a DynamoDb region OR Endpoint with signing region");
+                        Console.WriteLine(HelpText.AutoBuild(cmdOptions, null, null));
+                        return;
+                    }
+
+                    if (string.IsNullOrEmpty(options.DynamodbEndpoint) ||
+                        string.IsNullOrEmpty(options.DynamodbSigningRegion))
+                    {
+                        logger.LogError("One or more parameter(s) for endpoint configuration missing.");
+                        Console.WriteLine(HelpText.AutoBuild(cmdOptions, null, null));
+                        return;
+                    }
+
+                    builder.WithEndPointConfiguration(options.DynamodbEndpoint, options.DynamodbSigningRegion);
+                }
```

Comment:
This validation is a bit cumbersome and potentially confusing. If the `DynamodbRegion` and `DynamodbSigningRegion` flags are mutually exclusive yet both used to provide a region name for different config options, can we just combine them into a single flag, i.e., `DynamodbRegion`?

Then we could simplify with something like this:
```suggestion
                if (!string.IsNullOrEmpty(options.DynamodbEndpoint))
                {
                    if (string.IsNullOrEmpty(options.DynamodbRegion))
                    {
                        // TODO: check if region can be determined from provided endpoint
                        logger.LogError("DynamoDb region is required when providing a DynamoDb endpoint.");
                        Console.WriteLine(HelpText.AutoBuild(cmdOptions, null, null));
                        return;
                    }

                    builder.WithEndPointConfiguration(options.DynamodbEndpoint, options.DynamodbRegion);
                }
                else if (!string.IsNullOrEmpty(options.DynamodbRegion))
                {
                    builder.WithRegion(options.DynamodbRegion);
                }
```